### ISSUE TITLE
`garage` incorrect dependency resolution

### ISSUE THREAD

**Issue by danjenson** [python-poetry/poetry#6074]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Void Linux 5.18.14_1
- **Poetry version**: 1.1.14
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: https://gist.github.com/danjenson/02bb3e527480646019e11e6e03079f11

## Issue
Using the above `pyproject.toml` and running `poetry add garage` results in the following error:
![image](https://user-images.githubusercontent.com/4793438/180903934-83949439-89a3-4ae8-8563-15f3b33f4507.png)
However, the python version is 3.10, so this line in [setup.py](https://github.com/rlworkgroup/garage/blob/3492f446633a7e748f2f79077f6301c5b3ec9281/setup.py#L30) for `garage` should not be required.

**Comment by finswimmer** [comment]
Hello @danjenson,

the problem with this `setup.py` is, that is dynamically creates the list of dependencies. I guess you've installed Poetry with a Python Version <3.7. Thus, when Poetry gather the requirements it builds the wheel using this Python version and now `dataclasses` is included as a dependency.

Please contact the maintainer of this package and ask them to use environment markers to define when `dataclasses` should be a dependency. See https://peps.python.org/pep-0496/

It has to look like this:

```
dataclasses==0.7 ; python_version < '3.7'
```

So in summary, there is nothing Poetry can do here. It has to be fixed upstream.

fin swimmer
